source/credentials_provider_ecs.c

```diff
@@ -98,12 +103,33 @@ static struct aws_credentials_provider_ecs_user_data *s_aws_credentials_provider
         goto on_error;
     }
 
-    return wrapped_user_data;
+    struct aws_credentials_provider_ecs_impl *impl = ecs_provider->impl;
+    if (impl->auth_token_file_path) {
```

Comment:
nit: lets be consistent with how null checks are done? i've seen full != NULL in other places